cylc/flow/workflow_files.py

```diff
@@ -862,16 +862,22 @@ def check_reserved_dir_names(name: Union[Path, str]) -> None:
             raise WorkflowFilesError(err_msg.format('run<number>'))
 
 
-def infer_latest_run_from_id(workflow_id: str) -> str:
-    run_dir = Path(get_workflow_run_dir(workflow_id))
-    _, id_ = infer_latest_run(run_dir)
+def infer_latest_run_from_id(
+    workflow_id: str, alt_run_dir: Optional[str] = None
+) -> str:
+    if alt_run_dir is not None:
+        run_dir = Path(alt_run_dir) / Path(workflow_id)
```

Comment:
Can you do this with `Path`?

Reply:
Actually yes, cool..

Reply:
More usual to do this though
```suggestion
        run_dir = Path(alt_run_dir, workflow_id)
```